### DISCUSSION TITLE
Is there a way to create a schema without referring to a class or object?

### DISCUSSION THREAD

**Discussion by quinnjaymeo** [victools/jsonschema-generator#191] [General]
Hi, 

Great library!

I'm building an application that is based on OWL/rdfs and would like to create schemas dynamically, i.e., not based on predifined classes in my codebase. 

Is this possible? 

Thanks in advance!

**Comment by CarstenWickner** [comment]
Hi & Thanks,

From the top of my head I'd assume an intermediate step would be required that generates Java classes from the RDF descriptions.
Those generated classes could be fed to the schema generator and if you don't need them for anything else, you could just discard the generated classes afterwards.

E.g. put all generated classes in an otherwise unused package and then use reflection (as done in our Maven plugin here) to dynamically look up all classes in that package in order to produce JSON Schema definitions for them.

**Reply by quinnjaymeo** [reply]
Thanks for the quick response! I'll give it a shot.
